cpp/src/arrow/extension_type.h

```diff
@@ -43,7 +43,7 @@ class ARROW_EXPORT ExtensionType : public DataType {
   static constexpr const char* type_name() { return "extension"; }
 
   /// \brief The type of array used to represent this extension type's data
-  const std::shared_ptr<DataType>& storage_type() const { return storage_type_; }
+  std::shared_ptr<DataType> storage_type() const override { return storage_type_; }
```

Comment:
Why not also override `storage_type_ref`?

Reply:
I intended to remove storage_type_ref altogether